lib/src/modules/task.cc

```diff
@@ -128,11 +149,145 @@ static TaskCommand getTaskCommand(const fs::path &task_executable)
     return TaskCommand { task_executable.string(), { } };
 }
 
+// Converts boost filesystem errors to a task_error object.
+static Module::ProcessingError toModuleProcessingError(const fs::filesystem_error& e) {
+    std::string err_msg = e.what();
+    if (alg::contains(err_msg, "No such file or directory")) {
```

Comment:
Could we use `e.code()` defined in the [system_error class](http://www.boost.org/doc/libs/1_64_0/libs/system/doc/reference.html#Class-system_error) rather than matching against the message text?